dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java

```diff
@@ -890,19 +901,17 @@ private DynamicConfiguration prepareEnvironment(ConfigCenterConfig configCenter)
             DynamicConfiguration dynamicConfiguration;
             try {
                 dynamicConfiguration = getDynamicConfiguration(configCenter.toUrl());
-            } catch (Exception e) {
-                if (!configCenter.isCheck()) {
+                if (!dynamicConfiguration.isAvailable()) {
                     logger.warn(
-                            CONFIG_FAILED_INIT_CONFIG_CENTER,
-                            "",
-                            "",
-                            "The configuration center failed to initialize",
-                            e);
-                    configCenter.setInitialized(false);
-                    return null;
-                } else {
-                    throw new IllegalStateException(e);
+                            CONFIG_FAILED_INIT_CONFIG_CENTER, "", "", "The configuration center failed to initialize");
+                    if (!configCenter.isCheck()) {
+                        configCenter.setInitialized(true);
```

Comment:
1. ```initialized``` was already set to true at ```configCenter.checkOrUpdateInitialized(true)```.
2. you might miss the process when dynamicConfiguration.isAvailable() return false and configCenter.isCheck() returns true.